frontends/PyCDE/src/pycde/types.py

```diff
@@ -31,3 +31,14 @@ def struct(members) -> hw.StructType:
 
 
 types = _Types()
+
+
+def dim(inner_type_or_bitwidth, *size: int) -> hw.ArrayType:
+  """Creates a multidimensional array from innermost to outermost dimension."""
+  if isinstance(inner_type_or_bitwidth, int):
+    ret = types.int(inner_type_or_bitwidth)
+  else:
+    ret = inner_type_or_bitwidth
+  for s in size:
```

Comment:
should it go through `size` in order or reversed? i'm honestly not sure which makes more sense. as it stands, you specify the dimensions from innermost on the left to outermost on the right. my initial reaction was this is unintuitive, but that's because i'm used to reading IR from left to right. but i think in system verilog, it's also the outermost on the left down to the innermost on the right. what would `dim(6, 10, 12)` look like in system verilog? `logic [5:0][12][10] myArray`? if so, i think it makes sense to reverse `size` before iterating through it.

Reply:
There's nothing special about packed dimensions vs. integers, so it'd be `logic [11:0][9:0][5:0]` (or at least that's what I was shooting for). So while it is reversed from SV, I personally find it more intuitive.

Reply:
Agreed, it's SV that (as usual) is unintuitive